src/feature/nodelist/dirlist.c

```diff
@@ -49,6 +49,31 @@ static smartlist_t *trusted_dir_servers = NULL;
  * and all fallback directory servers. */
 static smartlist_t *fallback_dir_servers = NULL;
 
+/** Return true iff the given v4 address is the one of a configured directory
+ * authority matching the given type.
+ *
+ * This needs to be fast because it can be called at each directory connection
+ * in order to learn if the request is from an authority or not. */
```

Comment:
I think these issues are ok, but they need to be talked about in the function comments, in case they cause bugs in future:

You're checking the hard-coded (or configured) list of directory authorities, rather than the current consensus. That's probably ok, but it requires the directory authorities to update their configurations when the set of authority addresses changes. (Usually, if authorities advertise a new address in their descriptors, it gets picked up in the consensus, and used by other directory authorities, without a local config change. But this function is an exception to that rule.)

And you're checking the inbound ORPort address, rather than the OutboundBindAddress. How many directory authorities have different inbound and outbound addresses?

Reply:
Right so dirauth do have each authority added to their `torrc` at the moment. But we shouldn't rely on that for sure.

I'm confused here about the `OutboundBindAddress` comment. The code is using the address from the `dir_server_t` so if a dirauth uses a different address for outbound connection instead of the one advertised for inbound, we have a problem indeed.